File: apps/api/internal/domain/app_info.go
package domain

import "context"

// Kunci konfigurasi identitas branding aplikasi. Nama PT bukan kunci di sini:
// sumbernya tabel bank_profile (bank_name) agar hanya ada satu sumber kebenaran,
// sama seperti dokumen cetak. Kunci-kunci ini hanya untuk branding yang khas
// aplikasi, bukan identitas badan hukum.
const (
	ConfigKeyBrandingDisplayName = "branding.display_name"
	ConfigKeyBrandingShortName   = "branding.short_name"
	ConfigKeyBrandingDescription = "branding.description"
	ConfigKeyBrandingLogoURL     = "branding.logo_url"
)

// Nilai bawaan branding mereproduksi tampilan sebelum identitas menjadi dinamis.
// Dipakai hanya bila kunci tidak ada/rusak; nilai yang di-seed migrasi 000075
// sama dengan nilai ini sehingga tidak ada perubahan visual tak sengaja.
const (
	DefaultBrandingDisplayName = "CBS Core Backoffice"
	DefaultBrandingShortName   = "CBS Core"
	DefaultBrandingDescription = "Core Banking & Akuntansi Double-Entry"
)

// AppInfo adalah identitas aplikasi yang boleh dibaca TANPA login: dipakai halaman
// login dan metadata/judul tab web. Isinya sengaja dibatasi pada identitas publik;
// jangan menambah data internal (mis. daftar buku, konfigurasi, statistik) ke sini.
type AppInfo struct {
	// CompanyName adalah nama PT dari bank_profile.bank_name. Kosong berarti bank
	// belum mengisi profil; pemanggil tidak boleh mengarang nama.
	CompanyName string `json:"company_name"`
	// DisplayName adalah nama aplikasi yang tampil di header dan judul tab.
	DisplayName string `json:"display_name"`
	// ShortName adalah nama pendek untuk ruang sempit (mis. ikon/tab).
	ShortName string `json:"short_name"`
	// Description adalah keterangan singkat aplikasi.
	Description string `json:"description"`
	// LogoURL adalah URL logo; kosong berarti web memakai ikon bawaannya.
	LogoURL string `json:"logo_url"`
}

// AppInfoService menyusun identitas aplikasi untuk endpoint publik.
type AppInfoService interface {
	Get(ctx context.Context) AppInfo
}

File: apps/web/src/lib/app-info.ts
import type { AppInfo } from "./types";

/**
 * Identitas aplikasi bawaan. Dipakai hanya bila endpoint /api/v1/app-info belum
 * terjangkau (mis. saat build tanpa API). Nilainya sengaja sama dengan nilai seed
 * migrasi 000075 supaya tidak ada perubahan visual tak sengaja.
 */
export const DEFAULT_APP_INFO: AppInfo = {
  company_name: "",
  display_name: "CBS Core Backoffice",
  short_name: "CBS Core",
  description: "Core Banking & Akuntansi Double-Entry",
  logo_url: "",
};

/**
 * Menggabungkan identitas dari server dengan bawaan, agar field yang belum ada di
 * respons tidak menjadi `undefined` di UI.
 */
export function mergeAppInfo(
  data: Partial<AppInfo> | null | undefined,
): AppInfo {
  if (!data) return DEFAULT_APP_INFO;
  return {
    company_name: data.company_name ?? DEFAULT_APP_INFO.company_name,
    display_name: data.display_name ?? DEFAULT_APP_INFO.display_name,
    short_name: data.short_name ?? DEFAULT_APP_INFO.short_name,
    description: data.description ?? DEFAULT_APP_INFO.description,
    logo_url: data.logo_url ?? DEFAULT_APP_INFO.logo_url,
  };
}
